cicecore/cicedynB/analysis/ice_diagnostics_bgc.F90

```diff
@@ -937,19 +937,18 @@ subroutine zsal_diags
                enddo
                if (aice(i,j,iblk) > c0) &
                   psice_rho(n) = psice_rho(n)/aice(i,j,iblk)
-               if (tr_brine .and. aice(i,j,iblk) > c0) &
+               if (tr_brine .and. aice(i,j,iblk) > c0) then
                   phinS(n) = trcr(i,j,nt_fbri,iblk)*vice(i,j,iblk)/aice(i,j,iblk)
-
-               if (aicen(i,j,1,iblk)> c0) then
-                  if (tr_brine) phinS1(n) = trcrn(i,j,nt_fbri,1,iblk) &
-                                          * vicen(i,j,1,iblk)/aicen(i,j,1,iblk)
+                  phbrn(n) = (c1 - rhosi/rhow)*vice(i,j,iblk)/aice(i,j,iblk) &
```

Comment:
Should this account for variable snow density when the tracer is turned on?

Reply:
Possibly.  The changes that I made to this section of the code were only refactoring for better performance and shouldn't affect the overall logic or physics/diagnostics as previously coded at all.  There are quite a lot of places in the code where variable snow density could be used instead of rhos, beyond the new snow physics routines.  That's a whole new science project.